src/responder/nss/nss_iface.c

```diff
@@ -222,13 +222,15 @@ sss_nss_register_backend_iface(struct sbus_connection *conn,
     SBUS_INTERFACE(iface,
         sssd_nss_MemoryCache,
         SBUS_METHODS(
-            SBUS_SYNC(METHOD, sssd_nss_MemoryCache, UpdateInitgroups, sss_nss_memorycache_update_initgroups, nss_ctx),
-            SBUS_SYNC(METHOD, sssd_nss_MemoryCache, InvalidateAllUsers, sss_nss_memorycache_invalidate_users, nss_ctx),
-            SBUS_SYNC(METHOD, sssd_nss_MemoryCache, InvalidateAllGroups, sss_nss_memorycache_invalidate_groups, nss_ctx),
-            SBUS_SYNC(METHOD, sssd_nss_MemoryCache, InvalidateAllInitgroups, sss_nss_memorycache_invalidate_initgroups, nss_ctx),
-            SBUS_SYNC(METHOD, sssd_nss_MemoryCache, InvalidateGroupById, sss_nss_memorycache_invalidate_group_by_id, nss_ctx)
+            SBUS_SYNC(METHOD, sssd_nss_MemoryCache, UpdateInitgroups,
+                      sss_nss_memorycache_update_initgroups, nss_ctx)
+        ),
+        SBUS_SIGNALS(
+            SBUS_EMITS(sssd_nss_MemoryCache, InvalidateAllUsers),
+            SBUS_EMITS(sssd_nss_MemoryCache, InvalidateAllGroups),
+            SBUS_EMITS(sssd_nss_MemoryCache, InvalidateAllInitgroups),
+            SBUS_EMITS(sssd_nss_MemoryCache, InvalidateGroupById)
```

Comment:
I'm confused by terminology here.

'sssd_nss_MemoryCache' interface doesn't "emit" but "listens to (handles)" those signals, right?

Reply:
Yes, this should go away. Done.